lib/jsdom/living/helpers/geometry/matrixDomStringParser.js

```diff
@@ -0,0 +1,397 @@
+"use strict";
+/**
+ * Expected symbols when parsing dom string -> transform function
+ */
+const SYMBOL = {
+  number: /\d/,
+  char: /\D/,
+  lparen: /\(/,
+  rparen: /\)/,
+  whitespace: /\s/,
+  radixPoint: /\./,
+  comma: /,/,
+  commentOpen: /\/\*/,
+  commentClose: /\*\//
+};
+
+/**
+ * A very slim css-parser for extracting functions from a css dom string
+ */
+class CssFunctionParser {
```

Comment:
Please look over the spec for how to parse a CSS "component value" (https://drafts.csswg.org/css-syntax-3/#consume-component-value) and use that to implement this parser. Let's try to make it as spec-compliant as we can make it.

The syntax spec has two representations of a grammar construct: a railroad diagram, and a set of parsing algorithms. Always use the parsing algorithms: the railroad diagram is only for informative purposes.